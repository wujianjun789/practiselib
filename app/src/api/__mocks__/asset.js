/**
 * Created by a on 2017/7/26.
 */
import {HOST_IP, getHttpHeader, httpRequest} from '../../util/network'

export function getAssetModelList(cb) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/model-summaries', {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}

/**
 * 获取 model summary 通过 model ID
 * @param {String} modelID
 * @param {Function} cb
 * @return {Object}
 */
export function getModelSummariesByModelID(modelID, cb) {
	let response;
	const sensorList = [];
	switch(modelID) {
		case 'sensor':
			response = sensorList;
	}

	cb && cb(response);
}

export function getAssetList(cb) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/assets',{
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}

export function getSearchAssets(domainId, model, name, offset, limit, cb) {
	let response = [];
	cb && cb(response);
}

export function getSearchCount(domainId, model, name, cb) {
	const response = {count: 0};
	cb && cb(response);
}

function getSearchParam(domainId, model, name) {
    let param = {}
    if(domainId){
        Object.assign(param, {"domainId":domainId})
    }
    if(model){
        Object.assign(param, {"extendType":model})
    }
    if(name){
        Object.assign(param, {"name":{"like":name}})
    }

    return param;
}

export function getAssetsCount(cb) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/assets/count', {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}

/**
 *
 * @param model(资产模型类型)
 */
export function getAssetsByModel(model, cb){
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/'+model+'s', {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response)
    })
}

export function getAssetsBaseByModel(model, cb){
    let headers = getHttpHeader();

    let paramStr = JSON.stringify({"where":getSearchParam('', model, "")})

    httpRequest(HOST_IP+'/assets?filter='+encodeURIComponent(paramStr), {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}

export function getAssetsBaseById(id, cb) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/assets/'+id,{
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}
/**
 * 通过设备种类和域获取设备基本属性
 * @param model
 * @param name
 * @param cb
 */
export function getAssetsBaseByModelWithDomain(model, domainId, cb){
    let headers = getHttpHeader();

    let paramStr = JSON.stringify({"where":getSearchParam(domainId, model, "")})

    httpRequest(HOST_IP+'/assets?filter='+encodeURIComponent(paramStr), {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}
/**
 *
 * @param model(资产模型类型)
 */
export function getAssetsCountByModel(model, cb){
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/'+model+'s/count', {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response)
    })
}

/**
 * 通过设备种类和名称获取设备基本属性
 * @param model
 * @param name
 * @param cb
 */
export function getAssetsBaseByModelWithName(model, name, cb) {
    let headers = getHttpHeader();
    let paramStr = JSON.stringify({"where":getSearchParam("", model, name)})
    httpRequest(HOST_IP+'/assets?filter='+encodeURIComponent(paramStr), {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}
/**
 * param data({modelId:设备型号})
 *
 */
export function postAssetsByModel(model, data, cb){
    let headers = getHttpHeader();
    let dat = {id:data.id, type:data.modelId, base:{name:data.name, geoPoint:{lat:data.lat, lng:data.lng}, extendType:model, domainId:data.domainId}};
    httpRequest(HOST_IP+'/'+model+'s', {
        headers: headers,
        method: 'POST',
        body:JSON.stringify(dat)
    }, response=>{
        cb && cb(response);
    })
}

/**
 * param data({modelId:设备型号})
 *
 */
export function postXes(model, data, cb){
    let headers = getHttpHeader();
    let dat = {id:data.id, type:data.modelId, base:{name:data.name, geoPoint:{lat:data.lat, lng:data.lng}, extendType:model, domainId:data.domainId}};
    httpRequest(HOST_IP+'/xes', {
        headers: headers,
        method: 'POST',
        body:JSON.stringify(dat)
    }, response=>{
        cb && cb(response);
    })
}

/**
 * param data({modelId:设备型号})
 *
 */
export function updateAssetsByModel(model, data, cb) {
    let headers = getHttpHeader();
    let dat = {type:data.modelId};
    httpRequest(HOST_IP+'/'+model+'s/'+data.id, {
        headers: headers,
        method: 'PATCH',
        body:JSON.stringify(dat)
    }, response=>{
        updateAssetsById(response.id, {name:data.name, geoPoint:{lat:data.lat, lng:data.lng}, extendType:model, domainId:data.domainId}, cb)
    })
}

export function updateXes(model,data, cb){
    let headers = getHttpHeader();
    let dat = {
        "geoType":0,
        name:data.name,
        geoPoint:{
            lat:data.lat,
            lng:data.lng
        },
        extendType:model,
        domainId:data.domainId
    };
    httpRequest(HOST_IP+'/assets/'+data.id,{
        headers: headers,
        method: 'PATCH',
        body:JSON.stringify(dat)
    }, response=>{
        cb && cb(response);
    })
}

export function delAssetsByModel(model, id, cb) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/'+model+'s/'+id, {
        headers: headers,
        method: 'DELETE'
    }, response=>{
        cb && cb(response);
    })
}

export function delXes(model, id, cb) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/xes/'+id, {
        headers: headers,
        method: 'DELETE'
    }, response=>{
        cb && cb(response);
    })
}

export function updateAssetsById(id, data, cb){
    let headers = getHttpHeader();
    let dat = Object.assign({}, {"geoType":0}, data)
    httpRequest(HOST_IP+'/assets/'+id,{
        headers: headers,
        method: 'PATCH',
        body:JSON.stringify(dat)
    }, response=>{
        cb && cb(response);
    })
}

export function updateDataOrigin(data,key){

}

/**
 * 获取设备的状态
 * @param {String} model
 * @param {String | Number} id
 */
export const getDeviceStatusByModelAndId = (model, id) => cb => {
	const response = {};
	cb && cb(response);
}
