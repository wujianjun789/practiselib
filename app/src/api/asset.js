/**
 * Created by a on 2017/7/26.
 */
import { HOST_IP, getHttpHeader, httpRequest } from '../util/network';

export function getAssetModelList(cb) {
  let headers = getHttpHeader();
  httpRequest(HOST_IP + '/summaries', {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

/**
 * 获取 model summary 通过 model ID
 * @param {String} modelID
 * @param {Function} cb
 * @return {Object}
 */
export function getModelSummariesByModelID(modelID, cb) {
  httpRequest(`${HOST_IP}/summaries/${modelID}`, {
    headers: getHttpHeader(),
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

/**
 * 获取模型型号
 * @param modelId
 * @param cb
 */
export function getModelTypeByModel(modelId, cb) {
  const headers = getHttpHeader();
  httpRequest(HOST_IP + '/summaries/'+modelId+'/types',{
    headers: headers,
    method: 'GET'
  }, response=>{
    cb && cb(response);
  })
}

/**
 * 更新模型型号
 * @param modelId
 * @param data
 * @param cb
 */
export function updateModelTypeByModel(modelId, data, cb) {
  const headers = getHttpHeader();
  httpRequest(HOST_IP + '/summaries/'+modelId+'/types',{
    headers: headers,
    method: 'PUT',
    body:JSON.stringify(data)
  }, response=>{
    cb && cb(response);
  })
}

export function getAssetList(cb) {
  let headers = getHttpHeader();
  httpRequest(HOST_IP + '/assets', {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

export function getSearchAssets(domainId, model, name, offset, limit, cb) {
  let headers = getHttpHeader();
  let paramStr = JSON.stringify({ 'include': ['extend'], 'where': getSearchParam(domainId, model, name), 'offset': offset, 'limit': limit });
  httpRequest(HOST_IP + '/assets?filter=' + encodeURIComponent(paramStr), {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

export function getSearchCount(domainId, model, name, cb) {
  let headers = getHttpHeader();

  let paramStr = JSON.stringify(getSearchParam(domainId, model, name));

  httpRequest(HOST_IP + '/assets/count?where=' + encodeURIComponent(paramStr), {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

function getSearchParam(domainId, model, name) {
  let param = {};
  if (domainId) {
    Object.assign(param, { 'domainId': domainId });
  }
  if (model) {
    Object.assign(param, { 'extendType': model });
  }
  if (name) {
    Object.assign(param, { 'name': { 'like': name } });
  }

  return param;
}

export function getAssetsCount(cb) {
  let headers = getHttpHeader();
  httpRequest(HOST_IP + '/assets/count', {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

/**
 *
 * @param model(资产模型类型)
 */
export function getAssetsByModel(model, cb) {
  let headers = getHttpHeader();
  httpRequest(HOST_IP + '/' + model + 's', {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

export function getAssetsBaseByModel(model, cb) {
  let headers = getHttpHeader();

  let paramStr = JSON.stringify({ 'where': getSearchParam('', model, '') });

  httpRequest(HOST_IP + '/assets?filter=' + encodeURIComponent(paramStr), {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

export function getAssetsBaseById(id, cb) {
  let headers = getHttpHeader();
  httpRequest(HOST_IP + '/assets/' + id, {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}
/**
 * 通过设备种类和域获取设备基本属性
 * @param model
 * @param name
 * @param cb
 */
export function getAssetsBaseByModelWithDomain(model, domainId, cb) {
  let headers = getHttpHeader();

  let paramStr = JSON.stringify({ 'where': getSearchParam(domainId, model, '') });

  httpRequest(HOST_IP + '/assets?filter=' + encodeURIComponent(paramStr), {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

export function getAssetsBaseByDomain(domainId, cb) {
  let headers = getHttpHeader();
  httpRequest(HOST_IP + '/assets?filter=' + encodeURIComponent(JSON.stringify({ 'where': { 'domainId': domainId } })), {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}
/**
 *
 * @param model(资产模型类型)
 */
export function getAssetsCountByModel(model, cb) {
  let headers = getHttpHeader();
  httpRequest(HOST_IP + '/' + model + 's/count', {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

/**
 * 通过设备种类和名称获取设备基本属性
 * @param model
 * @param name
 * @param cb
 */
export function getAssetsBaseByModelWithName(model, name, cb) {
  let headers = getHttpHeader();
  let paramStr = JSON.stringify({ 'where': getSearchParam('', model, name) });
  httpRequest(HOST_IP + '/assets?filter=' + encodeURIComponent(paramStr), {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}
/**
 * param data({modelId:设备型号})
 *
 */
export function postAssetsByModel(model, data, cb) {
  let headers = getHttpHeader();
  let dat = { id: data.id, type: data.modelId, base: { name: data.name, geoPoint: { lat: data.lat, lng: data.lng }, extendType: model, domainId: data.domainId } };
  httpRequest(HOST_IP + '/' + model + 's', {
    headers: headers,
    method: 'POST',
    body: JSON.stringify(dat),
  }, response => {
    cb && cb(response);
  });
}

/**
 * param data({modelId:设备型号})
 *
 */
export function postXes(model, data, cb) {
  let headers = getHttpHeader();
  let dat = { id: data.id, type: data.modelId, base: { name: data.name, geoPoint: { lat: data.lat, lng: data.lng }, extendType: model, domainId: data.domainId } };
  httpRequest(HOST_IP + '/xes', {
    headers: headers,
    method: 'POST',
    body: JSON.stringify(dat),
  }, response => {
    cb && cb(response);
  });
}

/**
 * param data({modelId:设备型号})
 *
 */
export function updateAssetsByModel(model, data, cb) {
  let headers = getHttpHeader();
  let dat = { type: data.modelId };
  httpRequest(HOST_IP + '/' + model + 's/' + data.id, {
    headers: headers,
    method: 'PATCH',
    body: JSON.stringify(dat),
  }, response => {
    updateAssetsById(response.id, { name: data.name, geoPoint: { lat: data.lat, lng: data.lng }, extendType: model, domainId: data.domainId }, cb);
  });
}

export function updateXes(model, data, cb) {
  let headers = getHttpHeader();
  let dat = {
    'geoType': 0,
    name: data.name,
    geoPoint: {
      lat: data.lat,
      lng: data.lng,
    },
    extendType: model,
    domainId: data.domainId,
  };
  httpRequest(HOST_IP + '/assets/' + data.id, {
    headers: headers,
    method: 'PATCH',
    body: JSON.stringify(dat),
  }, response => {
    cb && cb(response);
  });
}

export function delAssetsByModel(model, id, cb) {
  let headers = getHttpHeader();
  httpRequest(HOST_IP + '/' + model + 's/' + id, {
    headers: headers,
    method: 'DELETE',
  }, response => {
    cb && cb(response);
  });
}

export function delXes(model, id, cb) {
  let headers = getHttpHeader();
  httpRequest(HOST_IP + '/xes/' + id, {
    headers: headers,
    method: 'DELETE',
  }, response => {
    cb && cb(response);
  });
}

export function updateAssetsById(id, data, cb) {
  let headers = getHttpHeader();
  let dat = Object.assign({}, { 'geoType': 0 }, data);
  httpRequest(HOST_IP + '/assets/' + id, {
    headers: headers,
    method: 'PATCH',
    body: JSON.stringify(dat),
  }, response => {
    cb && cb(response);
  });
}

export function updateDataOrigin(data, key) {

}

export function getAssetsByDomainLevelWithCenter(domainLevel, map, model, cb) {

  let headers = getHttpHeader();
  let nearParam = { maxDistance: map.distance / 1000, unit: 'kilometers' };

  if (domainLevel == 1) {
    nearParam = {};
  }

  let param = { geoPoint: Object.assign({}, { near: map.center }, nearParam) };
  if (model) {
    param = Object.assign({}, param, { extendType: model });
  }

  httpRequest(HOST_IP + '/assets?filter=' + encodeURIComponent(JSON.stringify({ 'where': param })), {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

export function getSearchAssetsByDomainWithCenter(domain, map, model, name, offset, limit, cb) {
  const headers = getHttpHeader();
  let nearParam = { maxDistance: map.distance / 1000, unit: 'kilometers' };
  if(domain.level == 1){
      nearParam = {};
  }

  const param = { geoPoint: Object.assign({}, { near: map.center }, nearParam)};
  let paramStr = JSON.stringify({ 'include': ['extend'], 'where': Object.assign({}, param, getSearchParam(domain.id, model, name)), 'offset': offset, 'limit': limit });
  httpRequest(HOST_IP + '/assets?filter=' + encodeURIComponent(paramStr), {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}

export function getSearchAssetCountByDomainWithCenter(domain, map, model, name, cb) {
  const headers = getHttpHeader();
  let nearParam = { maxDistance: map.distance / 1000, unit: 'kilometers' };
  if(domain.level == 1){
      nearParam = {};
  }

  const param = { geoPoint: Object.assign({}, { near: map.center }, nearParam)};
  let paramStr = JSON.stringify(Object.assign({}, param, getSearchParam(domain.id, model, name)));
  httpRequest(HOST_IP + '/assets/count?where=' + encodeURIComponent(paramStr), {
    headers: headers,
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
}
/**
 * 获取设备的状态
 * @param {String} model
 * @param {String | Number} id
 */
export const getDeviceStatusByModelAndId = (model, id) => cb => {
  // if(model == 'xes') model = 'xe';
  httpRequest(`${HOST_IP}/${model}s/${id}/status`, {
    headers: getHttpHeader(),
    method: 'GET',
  }, response => {
    cb && cb(response);
  });
};

/**
 * 设备操作
 * @param {number | string } id
 * @param {object} data
 * @param {function} cb
 */
export const updateAssetsRpcById = (id, data, cb) => {
  httpRequest(`${HOST_IP}/assets/${id}/rpc`, {
    headers: getHttpHeader(),
    method: 'POST',
    body: JSON.stringify(data),
  }, response => {
    cb && cb(response);
  });
};


/**
 * 删除设备型号
 * @param {number | string } id
 * @param {function} cb
 */
export const deleteModalTypesById = (id, cb) => {
  // httpRequest(`${HOST_IP}/`, {
  //   headers: getHttpHeader(),
  //   method: 'DELETE',
  // }, response => {
  //   cb && cb();
  // });
  console.log(id);
  cb && cb();
};

/**
 * 添加设备型号
 * @param {number | string } id
 * @param {function} cb
 */
export const addModalTypesById = (id, data, cb) => {
  // httpRequest(`${HOST_IP}/`, {
  //   headers: getHttpHeader(),
  //   method: 'POST',
  //   body: JSON.stringify(data),
  // }, response => {
  //   cb && cb(response);
  // });
  cb && cb();
};