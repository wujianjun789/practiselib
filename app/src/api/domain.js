/**
 * Created by a on 2017/7/26.
 */
import {HOST_IP, getHttpHeader, httpRequest} from '../util/network';
import {DOMAIN_LEVEL} from '../common/util/index';

export function getDomainList(cb) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/domains', {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}

export function getParentDomainList(cb) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/domains?filter='+encodeURIComponent(JSON.stringify({"where":{level:{neq:DOMAIN_LEVEL}}})), {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response)
    })
}

export function getChildDomainList(cb) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/domains?filter='+encodeURIComponent(JSON.stringify({"where":{level:DOMAIN_LEVEL}})), {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response)
    })
}

export function getDomainByDomainLevelWithCenter(domainLevel, map, cb) {
    let headers = getHttpHeader();
    let nearParam = {maxDistance: map.distance/1000,unit:'kilometers'}
    if(domainLevel==1){
        nearParam = {};
    }

    httpRequest(HOST_IP+'/domains?filter='+encodeURIComponent(JSON.stringify({"where": {geoPoint: Object.assign({}, {near: map.center}, nearParam), level:domainLevel}})),{
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })

}

export function getDomainById(id,cb){
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/domains/'+id, {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}

export function getDomainListByName(domainName, offset, limit, cb) {
    let headers = getHttpHeader();
    let obj = {include:["parent"], "offset":offset,"limit":limit}
    if(domainName){
        obj = Object.assign({"where":{name:{like:domainName}}}, obj);
    }
    let param = JSON.stringify(obj);
    let url = HOST_IP+'/domains?filter='+encodeURIComponent(param);

    httpRequest(url, {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}

export function getDomainCountByName(domainName, cb) {
    let headers = getHttpHeader();
    let param = JSON.stringify({name:domainName})
    let url = '';
    if(domainName){
        url = HOST_IP+'/domains/count?where='+encodeURIComponent(param);
    }else{
        url = HOST_IP+'/domains/count';
    }

    httpRequest(url, {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}

export function addDomain(domain, cb, err) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/domains',{
        headers: headers,
        method: 'POST',
        body: JSON.stringify(domain)
    }, response=>{
        cb && cb(response)
    }, null, error=>{
        err && err(error);
    })
}

export function updateDomainById(domain, cb, err) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/domains/'+domain.id,{
        headers: headers,
        method: 'PATCH',
        body: JSON.stringify(domain)
    }, response=>{
        cb && cb(response)
    }, null, error=>{
        err && err(error);
    })
}

export function deleteDomainById(domainId, cb) {
    let headers = getHttpHeader();
    httpRequest(HOST_IP+'/domains/'+domainId,{
        headers: headers,
        method: 'DELETE',
    }, response=>{
        cb && cb(response)
    },null, error=>{

    }, null, "unresolved")
}

export function getDomainListByParentId(parentId, cb){
    let headers = getHttpHeader();
    let param = JSON.stringify({where:{parentId:parentId}})
    httpRequest(HOST_IP+'/domains?filter='+encodeURIComponent(param), {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(parentId, response)
    })
}

export function getRelatedDomainById(id,cb){
    let headers = getHttpHeader();
    let param = JSON.stringify({include:['children']});
    httpRequest(`${HOST_IP}/domains/${id}?filter=${encodeURIComponent(param)}`, {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}



export function getWhiteListById(id, cb) {
    let headers = getHttpHeader();
    httpRequest(`${HOST_IP}/gateways/${id}/whiteList`, {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}

/**
 * 
 * @param {Number} gatewayId 
 * @param {Number} lcId 
 * @param {Function} cb 
 */
export function addLcToWhiteListById(gatewayId, lcId, cb) {   
    let headers = getHttpHeader();
    httpRequest(`${HOST_IP}/assets/${lcId}`,{
        headers: headers,
        method: 'PATCH',
        body:JSON.stringify({'gatewayId':gatewayId})
    }, response=>{
        cb && cb(response)
    })
}

/**
 * 
 * @param {Number} gatewayId 
 * @param {Number} lcId 
 * @param {Function} cb 
 */
export function delLcFromWhiteListById(gatewayId, lcId, cb) {
    let headers = getHttpHeader();
    httpRequest(`${HOST_IP}/assets/${lcId}`,{
        headers: headers,
        method: 'PATCH',
        body:JSON.stringify({'gatewayId':null})
    }, response=>{
        cb && cb(response)
    })
}

export function requestWhiteListCountById(gatewayId, cb) {
    let headers = getHttpHeader();
    httpRequest(`${HOST_IP}/gateways/${gatewayId}/whiteList/count`, {
        headers: headers,
        method: 'GET',
    }, response=>{
        cb && cb(response)
    })
}