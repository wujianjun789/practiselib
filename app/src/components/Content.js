/**
 * Created by a on 2017/7/4.
 */
import React,{Component} from 'react';

export default class Content extends Component{
    constructor(props){
        super(props)
    }

    render(){
        const {className=""} = this.props;
        return <div className={"content "+className}>
            {this.props.children}
        </div>
    }
}