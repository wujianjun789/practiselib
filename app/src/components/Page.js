import React, { Component } from 'react';

import '../../public/styles/rc-pagination.css';
import Pagination from 'rc-pagination';
//分页器
export default class Page extends Component {
    constructor(props) {
        super(props)
        this.paginationOnChange = this.paginationOnChange.bind(this);
    }

    paginationOnChange(page) {
        this.props.onChange && this.props.onChange(page);
    }

    render() {
        let props = this.props;
        return (
            <Pagination {...props} onChange={ this.paginationOnChange } />
        )
    }
}