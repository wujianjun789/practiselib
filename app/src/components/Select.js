/**
 * Created by a on 2017/7/7.
 */
import React, { Component } from 'react'
import Immutable from 'immutable';

export default class Select extends Component {
    constructor(props) {
        super(props)
        this.onChange = this.onChange.bind(this);
    }

    onChange(event) {
        this.props.onChange && this.props.onChange(event.target.selectedIndex);
    }

    render() {
        const {className='', id='', valueKey="value", titleKey="value", data=Immutable.fromJS({
                list: [],
                value: '',
                placeholder: ""
            })} = this.props;

        return <select className={ "select " + className + " form-control" } id={ id } placeholder={ data.get('placeholder') } value={ data.get("value") } onChange={ this.onChange }>
                 { data.get('list').map((option, index) => {
                       let value = option.get(valueKey);
                       let title = option.get(titleKey);
                       return <option key={ index } value={ value }>
                                { title }
                              </option>
                   }) }
               </select>
    }
}