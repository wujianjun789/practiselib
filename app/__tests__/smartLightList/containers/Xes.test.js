/**
 * Created by Azrael on 2017/9/27
 */
jest.mock('../../../src/util/network.js');
jest.mock('../../../src/api/asset.js');
jest.mock('../../../src/api/domain.js');
import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Content from '../../../src/components/Content';
import SearchText from '../../../src/components/SearchText';
import Select from '../../../src/components/Select.1';
import TableWithHeader from '../../../src/smartLightList/components/TableWithHeader';
import TableTr from '../../../src/smartLightList/components/TableTr';
import Page from '../../../src/components/Page';
import Xes from '../../../src/smartLightList/containers/Xes';
import {getDomainList} from '../../../src/api/domain';
import {getSearchAssets, getSearchCount, getDeviceStatusByModelAndId} from '../../../src/api/asset';

describe('<Xes />', () => {
	it('default render', () => {
		const cmp = shallow(<Xes />);

		const content = cmp.find('Content');
		expect(!content.hasClass('collapse')).toBeTruthy();

		const select = cmp.find('#domain');
		const domainList_state = cmp.state('domainList');
		const currentDomain_state = cmp.state('currentDomain');
		Object.keys(domainList_state).forEach(item => {
			expect(select.prop(item)).toEqual(domainList_state[item]);
		});
		expect(select.prop('value')).toEqual(currentDomain_state == null ? '' : currentDomain_state[domainList_state.valueField]);

		const searchText = cmp.find('SearchText');
		const search_state = cmp.state('search');
		Object.keys(search_state).forEach(item => {
			expect(searchText.prop(item)).toBe(search_state[item]);
		});

		const tableWithHeader = cmp.find('TableWithHeader');
		const inst = cmp.instance();
		const columns = inst.columns;
		expect(tableWithHeader.prop('columns')).toEqual(columns);

		const tableTrs = cmp.find('TableTr');
		const deviceList_state = cmp.state('deviceList');
		const currentDevice_state = cmp.state('currentDevice');
		deviceList_state.forEach((item, index) => {
			let curTr = tableTrs.at(index);
			expect(curTr.key()).toBe(item.id);
			expect(curTr.prop('data')).toEqual(item);
			expect(curTr.prop('columns')).toEqual(columns);
			expect(curTr.prop('activeId')).toBe(currentDevice_state.id);
		});

		const page = cmp.find('Page');
		const page_state = cmp.state('page');
		expect(page.hasClass('hidden')).toBe(page_state.total == 0 ? true : false);
		expect(page.prop('pageSize')).toBe(page_state.limit);
		expect(page.prop('current')).toBe(page_state.current);
		expect(page.prop('total')).toBe(page_state.total);

		const sidebarInfo = cmp.find('.sidebar-info');
		expect(!sidebarInfo.hasClass('sidebar-collapse')).toBeTruthy();

		// 测试icon className
		expect(sidebarInfo.find('.icon_verital').length).toBe(1);

		const panel1 = sidebarInfo.find('.panel-1');
		expect(panel1.find('.panel-heading').text()).toBe('选中设备');
		expect(panel1.find('.panel-body span').text()).toBe(currentDevice_state == null ? '' : currentDevice_state.name);
		expect(panel1.find('.panel-body span').props().title).toBe(currentDevice_state == null ? '' : currentDevice_state.name);
	});

	it('simulate click', () => {
		const cmp = mount(<Xes />);
		let content = cmp.find('Content');
		expect(!content.hasClass('collapse')).toBeTruthy();
		content.find('.sidebar-info .collapse-container').simulate('click');
		content = cmp.find('Content');
		expect(content.hasClass('collapse')).toBeTruthy();
		expect(content.find('.sidebar-info').hasClass('sidebar-collapse')).toBeTruthy();
		expect(content.find('.sidebar-info .collapse-container span').hasClass('icon_horizontal')).toBeTruthy();
		content.find('.sidebar-info .collapse-container').simulate('click');
		content = cmp.find('Content');

		const domainList = {
			titleField: 'name',
			valueField: 'name',
			options: [{id: 1, name: '域一'},{id: 2, name: '域二'}]
		}
		cmp.setState({domainList});
		const _select = cmp.find('#domain');
		let event = {target: {id: 'domain', selectedIndex: 0}};
		_select.find('select').simulate('change', event);
		expect(_select.prop('value')).toBe(domainList.options[event.target.selectedIndex].name);

		const searchText = cmp.find('SearchText');
		event = {target: {value: '搜索'}};
		searchText.find('input').simulate('change', event);
		expect(searchText.prop('value')).toBe(event.target.value);

		const deviceList = [
			{id: 0, name: '设备1', online: true},
			{id: 1, name: '设备2', online: false}
		];
		const currentDevice = deviceList[0];
		cmp.setState({deviceList, currentDevice});
		const tableTrs = cmp.find('TableTr');
		tableTrs.at(1).find('tr').simulate('click');
		expect(cmp.state('currentDevice')).toEqual(deviceList[1]);
	});

	it('default render snapshot', () => {
		const cmp = renderer.create(<Xes />);
		expect(cmp.toJSON()).toMatchSnapshot();
	})
})
