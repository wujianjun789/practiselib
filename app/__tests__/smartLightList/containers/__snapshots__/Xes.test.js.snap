exports[`<Xes /> default render snapshot 1`] = `
<div
  className="content list-collect ">
  <div
    className="content-left">
    <div
      className="heading">
      <select
        className="form-control select "
        id="domain"
        onChange={[Function]}
        value="域一">
        <option
          value="域一">
          域一
        </option>
        <option
          value="域一">
          域一
        </option>
      </select>
      <div
        className="searchText "
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="请输入设备名称"
          type="search"
          value="" />
        <ul
          className="" />
        <span
          className="glyphicon glyphicon-search"
          onClick={[Function]} />
      </div>
    </div>
    <div
      className="table-container">
      <div
        className="table-responsive undefined">
        <table
          className="table table-hover">
          <thead>
            <tr>
              <th>
                设备名称
              </th>
              <th>
                在线状态
              </th>
              <th>
                故障状态
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      <ul
        className="rc-pagination page hidden"
        style={Object {}}
        unselectable="unselectable">
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-prev"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="上一页">
          <a />
        </li>
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="下一页">
          <a />
        </li>
        <div
          className="rc-pagination-options" />
      </ul>
    </div>
  </div>
  <div
    className="container-fluid sidebar-info ">
    <div
      className="row collapse-container"
      onClick={[Function]}>
      <span
        className="icon_verital" />
    </div>
    <div
      className="panel panel-default panel-1">
      <div
        className="panel-heading">
        <svg>
          <use
            height="200"
            transform="scale(0.08,0.08)"
            viewBox="0 0 20 20"
            width="200"
            x="0"
            xlinkHref="#icon_sys_select"
            y="0" />
        </svg>
        选中设备
      </div>
      <div
        className="panel-body">
        <span
          title="">
          
        </span>
      </div>
    </div>
  </div>
</div>
`;
